src/node/chainstate.h

```diff
@@ -31,14 +32,11 @@ struct ChainstateLoadOptions {
     std::function<void()> coins_error_cb;
 };
 
-//! Chainstate load status. Simple applications can just check for the success
-//! case, and treat other cases as errors. More complex applications may want to
-//! try reindexing in the generic failure case, and pass an interrupt callback
-//! and exit cleanly in the interrupted case.
-enum class ChainstateLoadStatus { SUCCESS, FAILURE, FAILURE_INCOMPATIBLE_DB, INTERRUPTED };
-
-//! Chainstate load status code and optional error string.
-using ChainstateLoadResult = std::tuple<ChainstateLoadStatus, bilingual_str>;
+//! Chainstate load errors. Simple applications can just treat all errors as
+//! failures. More complex applications may want to try reindexing in the
+//! generic error case, and pass an interrupt callback and exit cleanly in the
+//! interrupted case.
+enum class ChainstateLoadError { FAILURE, FAILURE_INCOMPATIBLE_DB, INTERRUPTED };
```

Comment:
Perhaps could take the opportunity to make this something like:

```suggestion
enum class ChainstateLoadError { FAILURE_TRY_REINDEX, FAILURE_FATAL, INTERRUPTED };
```

So that it's more inline with indicating "what to do about the failure".

Could also move the "4 types of outcomes" comment up here where it's clearer.